File: ListUtils.kt
// Example:
// var li = ListNode(5)
// var v = li.`val`
// Definition for singly-linked list.
class ListNode(var `val`: Int) {
    var next: ListNode? = null
}

object ListUtils {
    fun buildList(nums: IntArray?): ListNode? {
        if (nums == null) return null
        val dummy = ListNode(0)
        var p = dummy
        for (i in nums) {
            p.next = ListNode(i)
            p = p.next!!
        }
        return dummy.next
    }

    fun printList(head: ListNode?) {
        var head = head
        while (head != null) {
            System.out.printf("%d ", head.`val`)
            head = head.next
        }
        println()
    }
}

File: ListUtils.py


from typing import Optional


class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


def buildList(values) -> Optional[ListNode]:
    if values is None or len(values) == 0:
        return None
    head = ListNode(values[0])
    node = head
    for i in values[1:]:
        node.next = ListNode(i)
        node = node.next

    return head


def printList(head: Optional[ListNode]):
    if head is not None:
        print(head.val, end='')
        head = head.next
    while head is not None:
        print('->'+str(head.val), end='')
        head = head.next
    print()
